Marlin/planner.cpp

```diff
@@ -1013,26 +1015,42 @@ void Planner::_buffer_line(const float &a, const float &b, const float &c, const
   }
 
   // Compute and limit the acceleration rate for the trapezoid generator.
-  float steps_per_mm = block->step_event_count / block->millimeters;
+  float steps_per_mm = block->step_event_count * inverse_millimeters;
   uint32_t accel;
   if (!block->steps[X_AXIS] && !block->steps[Y_AXIS] && !block->steps[Z_AXIS]) {
     // convert to: acceleration steps/sec^2
     accel = ceil(retract_acceleration * steps_per_mm);
   }
   else {
-    #define LIMIT_ACCEL(AXIS) do{ \
-      if (max_acceleration_steps_per_s2[AXIS] < (accel * block->steps[AXIS]) / block->step_event_count) \
-        accel = (max_acceleration_steps_per_s2[AXIS] * block->step_event_count) / block->steps[AXIS]; \
+    #define LIMIT_ACCEL_32(AXIS) do{ \
+      if (block->steps[AXIS]) { \
+        const uint32_t comp = max_acceleration_steps_per_s2[AXIS] * block->step_event_count; \
+        if (accel * block->steps[AXIS] > comp) accel = comp / block->steps[AXIS]; \
+      } \
+    }while(0)
+
+    #define LIMIT_ACCEL_64(AXIS) do{ \
+      if (block->steps[AXIS]) { \
+        const uint64_t comp = (uint64_t)max_acceleration_steps_per_s2[AXIS] * (uint64_t)block->step_event_count; \
+        if ((uint64_t)accel * (uint64_t)block->steps[AXIS] > comp) accel = comp / (uint64_t)block->steps[AXIS]; \
+      } \
```

Comment:
Extended the test by 'long long'
```
...
  multiply byte             : 1.386 us
  divide byte               : 6.290 us
  add byte                  : 1.324 us
  multiply integer          : 2.141 us
  divide integer            : 15.217 us
  add integer               : 1.638 us
  multiply long             : 6.352 us
  divide long               : 38.887 us
  add long                  : 1.763 us
  multiply long long        : 23.337 us *
  divide long long          : 20.012 us **
  add long long             : 4.153 us *
  multiply float            : 7.742 us
  divide float              : 83.487 us
  add float                 : 9.602 us
...
```
Ether i did something wrong - or we have a solid surprise.